airflow/executors/kubernetes_executor.py

```diff
@@ -599,13 +599,14 @@ def sync(self) -> None:
                 try:
                     self.kube_scheduler.run_next(task)
                 except ApiException as e:
-                    if e.reason == "BadRequest":
-                        self.log.error("Request was invalid. Failing task")
+                    if e.reason in ("BadRequest", "Unprocessable Entity"):
+                        self.log.error(f"Pod creation failed with reason {e.reason!r}. Failing task")
```

Comment:
I prefer we use `e.status` for this and give different error messages for BadRequest error and 422. No strong opinion though

Reply:
It looks like we could do that:

```
        - 403 Forbidden will be returned when your request exceeds namespace quota.
        - 422 Unprocessable Entity is returned when your parameters are valid but unsupported
            e.g. limits lower than requests.
        - 400 BadRequest is returned when your parameters are invalid e.g. asking for cpu=100ABC123.
```

I guess one reason to use `reason` in the if statement though, is that it's english.  So we can read

```
if e.reason in ("BadRequest", "Unprocessable Entity"):
```

rather than

```
if e.status in (400, 422):
```

because the former is more descriptive / self-documenting.  

But I also don't have a strong opinion on this.

Reply:
```suggestion
                    if e.reason in ("BadRequest", "Unprocessable Entity"):
                        self.log.error("Pod creation failed with reason %r. Failing task", e.reason)
```
```suggestion
                    if e.reason in ("BadRequest", "Unprocessable Entity"):
                        self.log.error(f"Pod creation failed with reason {e.reason!r}. Failing task")
```

Reply:
One reason (uh) against using `reason` is it's more brittle. `reason` is intended as human readable and could potentially change (e.g. `BadRequest` to `Bad Request`), while the status code is for machines and more suitable for Airflow's use case here IMO.

Reply:
yeah i can see that point.  another reason i went with `reason` was ... well... the code was already using `reason` for `BadRequest`.  And you know it occurred to me the possibility that the status codes have less fidelity than the reasons, i.e. possibly less precise, so that if we were to change this to status code, perhaps we would catch more scenarios in this block than was originally intended. e.g if 400 can be `BadRequest` and `LessBadRequest`, we might start catching something we weren't supposed to.

but, i've seen no evidence of this, and since the majority seems to be in favor of status code, i'll make the change.

Reply:
and i've updated it, ptal